src/plugins/content_management/server/core/core.test.ts

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { loggingSystemMock } from '@kbn/core/server/mocks';
+import type { ContentConfig } from './types';
+import { Core } from './core';
+import { createMemoryStorage, MockContent } from './mocks';
+import { ContentRegistry } from './registry';
+import { ContentCrud } from './crud';
+import type {
+  GetItemStart,
+  GetItemSuccess,
+  GetItemError,
+  CreateItemStart,
+  CreateItemSuccess,
+  CreateItemError,
+  UpdateItemStart,
+  UpdateItemSuccess,
+  UpdateItemError,
+  DeleteItemStart,
+  DeleteItemSuccess,
+  DeleteItemError,
+} from './event_types';
+import { EventBus } from './event_bus';
+
+const logger = loggingSystemMock.createLogger();
+
+describe('Content Core', () => {
+  describe('setup()', () => {
+    let core: Core;
+    let coreSetup: ReturnType<Core['setup']>;
+    let contentConfig: ContentConfig;
+
+    beforeEach(() => {
+      core = new Core({ logger });
+      coreSetup = core.setup();
+      contentConfig = {
+        storage: createMemoryStorage(),
+      };
+    });
+
+    afterEach(() => {
+      // Remove subscription of the events$
+      coreSetup.api.eventBus.stop();
+    });
```

Comment:
nit: instead of Jest built-in `beforeEach` and `afterEach`, I like to use a helper function, curious what do you think about that?

Something like this:

```ts
test('...', async () => {
  const { contentCrud } = await setup(); // This call ensures there is no globals, everything is returned.

  // Test logic goes here.

  // No need for "afterEach", as there are no globals. Everything returned by "setup()" is just garbage collected.
});
```

Reply:
I like that approach. I refactor to use it https://github.com/elastic/kibana/pull/150692/commits/67fe43aa0548f04fe0146776ea805c710b125693

> No need for afteEach

Actually sometimes we need some cleanup after each test. I provided a `cleanUp` handler when returning from the `setup()` call.